relayer/chains/icon/provider.go

```diff
@@ -134,21 +135,26 @@ func (p *Provider) FinalityBlock(ctx context.Context) uint64 {
 
 // MessageReceived checks if the message is received
 func (p *Provider) MessageReceived(ctx context.Context, messageKey *providerTypes.MessageKey) (bool, error) {
-	callParam := p.prepareCallParams(MethodGetReceipts, p.cfg.Contracts[providerTypes.ConnectionContract], map[string]interface{}{
-		"srcNetwork": messageKey.Src,
-		"_connSn":    types.NewHexInt(messageKey.Sn.Int64()),
-	})
-
-	var status types.HexInt
-	if err := p.client.Call(callParam, &status); err != nil {
-		return false, fmt.Errorf("MessageReceived: %v", err)
+	switch messageKey.EventType {
+	case events.EmitMessage:
+		callParam := p.prepareCallParams(MethodGetReceipts, p.cfg.Contracts[providerTypes.ConnectionContract], map[string]interface{}{
+			"srcNetwork": messageKey.Src,
+			"_connSn":    types.NewHexInt(messageKey.Sn.Int64()),
+		})
+		var status types.HexInt
+		return status == types.NewHexInt(1), p.client.Call(callParam, &status)
+	case events.CallMessage:
+		return false, nil
+	case events.RollbackMessage:
+		return false, nil
+	default:
+		return true, fmt.Errorf("unknown event type")
 	}
-	return status == types.NewHexInt(1), nil
 }
 
 // ReverseMessage reverts a message
 func (p *Provider) RevertMessage(ctx context.Context, sn *big.Int) error {
-	params := map[string]interface{}{"_sn": types.NewHexInt(sn.Int64())}
+	params := map[string]interface{}{"sn": types.NewHexInt(sn.Int64())}
```

Comment:
same here, it seems to fix issue with revert call which is not associated with the rollback changes. It would be easier to track changes if it were made in different PR.